sympy/matrices/matrices.py

```diff
@@ -1594,6 +1596,12 @@ def __pow__(self, num):
                     return self._matrix_pow_by_recursion(num)
             return self._matrix_pow_by_recursion(num)
         elif isinstance(num, (Expr, float)):
+            num = sympify(num)
+            if (num >= 0) == True and not num % 1 == 0:
+                z = self.jordan_form()[1]
```

Comment:
```
 if (num >= 0) == True and not num % 1 == 0 and jc.rows > 1 and l == 0: 
 ```
So this condition must be sufficient in ```_matrix_pow_by_jordan_blocks``` to raise an error. After this I think there's no need to change anything in```__pow__```.